authorization-common/src/main/java/com/sequenceiq/authorization/utils/EventAuthorizationUtils.java

```diff
@@ -0,0 +1,99 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventAuthorizationUtils.class);
+
+    private final Map<AuthorizationResourceType, AuthorizationResourceAction> specials;
+
+    private CommonPermissionCheckingUtils permissionCheckingUtils;
+
+    public EventAuthorizationUtils(CommonPermissionCheckingUtils permissionCheckingUtils) {
+        this.permissionCheckingUtils = permissionCheckingUtils;
+        specials = new HashMap<>();
+        for (AuthorizationResourceAction resourceAction : Arrays.asList(AuthorizationResourceAction.values())) {
+            if (resourceAction.getAuthorizationResourceType() == CUSTOM_CONFIGURATIONS && resourceAction.isDescribeAction()) {
+                specials.put(CUSTOM_CONFIGURATIONS, resourceAction);
+            }
+        }
+    }
+
+    public void checkPermissionBasedOnResourceTypeAndCrn(Collection<EventAuthorizationDto> eventAuthorizationDtos) {
+        throwIfNull(eventAuthorizationDtos,
+                () -> new IllegalArgumentException("The collection of " + EventAuthorizationDto.class.getSimpleName() + "s should not be null!"));
+        LOGGER.info("Checking permissions for events: {}",
+                String.join(",", eventAuthorizationDtos.stream().map(EventAuthorizationDto::toString).collect(Collectors.toSet())));
+        for (EventAuthorizationDto dto : eventAuthorizationDtos) {
+            Set<AuthorizationResourceAction> act = getActionsForResourceType(findResourceTypeByString(dto.getResourceType()));
+            checkMultipleActions(act, dto.getResourceCrn());
+        }
+    }
+
+    private AuthorizationResourceType findResourceTypeByString(String resourceType) {
+        try {
+            return AuthorizationResourceType.valueOf(resourceType.toUpperCase());
+        } catch (NullPointerException | IllegalArgumentException iae) {
+            throw new IllegalStateException("Unable to find AuthZ action for resource: " + resourceType);
+        }
+    }
+
+    private boolean checkSpecialActionHasPermission(AuthorizationResourceAction action, String resourceCrn) {
+        for (AuthorizationResourceAction alternative : ALTERNATIVES.get(action)) {
+            try {
+                checkPermission(alternative, resourceCrn);
+                return true;
+            } catch (AccessDeniedException ade) {
+                LOGGER.info(AccessDeniedException.class.getSimpleName() + " has caught during - alternative - permission check for " +
+                        "[action: {}, resourceCrn: {}]", action.name(), resourceCrn);
+            }
+        }
+        return false;
+    }
+
+    private void checkMultipleActions(Collection<AuthorizationResourceAction> actions, String resourceCrn) {
+        for (AuthorizationResourceAction action : actions) {
+            if (ALTERNATIVES.containsKey(action)) {
+                if (checkSpecialActionHasPermission(action, resourceCrn)) {
+                    continue;
+                }
```

Comment:
I am not sure I got the logic here:
if method returns true, we will continue for loop, and we will do the same in case of method returns false, don't we?